chainerx_cc/chainerx/routines/linalg.cc

```diff
@@ -98,4 +98,83 @@ Array Dot(const Array& a, const Array& b, nonstd::optional<Dtype> out_dtype) {
     return out_matrix.Reshape(out_shape);
 }
 
+Array Solve(const Array& a, const Array& b) {
+    if (a.ndim() != 2) {
+        throw DimensionError{"ChainerX solve supports only 2-dimensional arrays."};
+    }
+    if (a.shape()[0] != a.shape()[1]) {
+        throw DimensionError{"Matrix is not square."};
+    }
+    CheckEqual(a.device(), b.device());
+    CheckEqual(a.dtype(), b.dtype());
```

Comment:
- How about throwing an error here in the routine if the dtype is not fp32 or fp64? I think supporting these two for now is okay. What do you think? (An alternative is to delay the dtype check and let the device kernel decide)
- Let's support mixed precision similar to NumPy, i.e. fp32 with fp64.

Reply:
I think dtype check should be on the level of device kernel since it's likely that GPU and CPU would have different coverage.